spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
             index = self._thumbnails.index(self.current_thumbnail) - 1
             index = index if index >= 0 else len(self._thumbnails) - 1
             self.set_current_index(index)
+            self.scroll_to_select_item(index)
 
     def go_next_thumbnail(self):
         """Select thumbnail next to the currently selected one."""
         if self.current_thumbnail is not None:
             index = self._thumbnails.index(self.current_thumbnail) + 1
             index = 0 if index >= len(self._thumbnails) else index
             self.set_current_index(index)
+            self.scroll_to_select_item(index)
+
+    def scroll_to_select_item(self, index):
+        """Scroll to the selected item of ThumbnailScrollBar."""
+        spacing_between_items = self.scene.verticalSpacing()
+        height_view = self.scrollarea.viewport().height()
+        height_item = self.scene.itemAt(index).sizeHint().height()
+        height_view_excluding_item = height_view - height_item
+        if height_view_excluding_item < 0:
+            height_view_excluding_item = 0
+
+        height_of_top_items = spacing_between_items
+        for i in range(index):
+            item = self.scene.itemAt(i)
+            height_of_top_items += item.sizeHint().height()
+            height_of_top_items += spacing_between_items
+
+        pos_scroll = height_of_top_items - height_view_excluding_item // 2
+
+        vsb = self.scrollarea.verticalScrollBar()
```

Comment:
Is there a chance that there is no scroll bar at all if there are only a few or couple of items?

Reply:
ThumbnailScrollBar creates a verticalScrollBar in __init__ method.

https://github.com/spyder-ide/spyder/blob/8442cadd0f6fec887e0297f8dfe29fcfa17120fa/spyder/plugins/plots/widgets/figurebrowser.py#L537